pkg/storage/allocator.go

```diff
@@ -444,7 +464,12 @@ func (a Allocator) RemoveTarget(
 // rebalance. This helps prevent a stampeding herd targeting an abnormally
 // under-utilized store.
 func (a Allocator) RebalanceTarget(
-	ctx context.Context, constraints config.Constraints, rangeInfo RangeInfo, filter storeFilter,
+	ctx context.Context,
+	constraints config.Constraints,
+	repl *Replica,
```

Comment:
Passing in a `Replica` is pretty broad. It looks like we could shrink this down to just using the `RangeInfo` if we made `simulateRemoveTarget` just take a `RangeInfo` instead of a `Replica`.

Reply:
So we should guarantee the duration for `WritesPerSecond` in `RangeInfo` exceeds `MinStatsDuration`?

Reply:
Yes, I suppose so. Since this code only runs on the leaseholder, it pretty much always should be, anyway.